Bump docker to version 11.0.4

gurkanindibay opened the pull request.

gurkanindibay commented on .github/workflows/update_version.yml:
```diff
@@ -8,9 +8,6 @@ on:
     inputs:
       project_version:
         description: "Version number to be released e.g 10.0.3"
-      postgres_version:
```
This one is additional. I had removed postgres version update long time ago. I removed here in the pipeline as well to be able to execute update_docker from Actions pipeline screen

hanefi reviewed (changes requested):
Changes related to version bump look good. However there are 2 changes that I'd like to see here.

1. Please remove git submodule for tools repo. It causes fatal warnings on builds.
2. Why do we update `postgres-12/Dockerfile` with a version that does not support PG12? I suggest we remove that file completely. We already removed it from `publish_docker_images_on_push` matrix so we already stopped building images using that on CI.

gurkanindibay commented:
> Changes related to version bump look good. However there are 2 changes that I'd like to see here.
> 
> 1. Please remove git submodule for tools repo. It causes fatal warnings on builds.
> 2. Why do we update `postgres-12/Dockerfile` with a version that does not support PG12? I suggest we remove that file completely. We already removed it from `publish_docker_images_on_push` matrix so we already stopped building images using that on CI.

1. Removed
2. https://github.com/citusdata/docker/issues/306 I will fix with that issue
